graylog2-web-interface/src/components/welcome/hooks.tsx

```diff
@@ -0,0 +1,112 @@
+} from 'components/welcome/types';
+
+export const urlPrefix = '/startpage';
+export const LAST_OPEN_QUERY_KEY = 'last_open_query_key';
+export const PINNED_ITEMS_QUERY_KEY = 'pinned_items_query_key';
+export const RECENT_ACTIONS_QUERY_KEY = 'recent_actions_query_key';
+
+const fetchLastOpen = async ({ page }: RequestQuery): Promise<PaginatedLastOpened> => {
+  const url = PaginationURL(`${urlPrefix}/lastOpened`, page, 5, '');
+
+  return fetch('GET', qualifyUrl(url));
+};
+
+const fetchFavoriteItems = async ({ page }: RequestQuery): Promise<PaginatedFavoriteItems> => {
+  const url = PaginationURL('/favorites', page, 5, '');
+
+  return fetch('GET', qualifyUrl(url));
+};
+
+const fetchRecentActivities = async ({ page }: RequestQuery): Promise<PaginatedRecentActivity> => {
+  const url = PaginationURL(`${urlPrefix}/recentActivity`, page, 5, '');
+
+  return fetch('GET', qualifyUrl(url)).then((data: PaginatedResponseRecentActivity): PaginatedRecentActivity => {
+    return ({
+      page: data.page,
+      per_page: data.per_page,
+      count: data.count,
+      total: data.total,
+      recentActivity: data.recentActivity.map((activity) => ({
+        id: activity.id,
+        itemTitle: activity.item_title,
+        timestamp: activity.timestamp,
+        activityType: activity.activity_type,
+        itemType: activity.item_type,
+        itemId: activity.item_id,
+        userName: activity.user_name,
+      })),
+    });
+  });
+};
+
+export const useLastOpened = (pagination: RequestQuery): { data: PaginatedLastOpened, isFetching: boolean } => {
+  return useQuery([LAST_OPEN_QUERY_KEY, pagination], () => fetchLastOpen(pagination), {
+    onError: (errorThrown) => {
+      UserNotification.error(`Loading last opened items failed with status: ${errorThrown}`,
+        'Could not load last opened items');
+    },
+    retry: 0,
+    initialData: {
+      lastOpened: [],
+      ...DEFAULT_PAGINATION,
+    },
+  });
+};
+
+export const useFavoriteItems = (pagination: RequestQuery): {data: PaginatedFavoriteItems, isFetching: boolean} => {
+  return useQuery([PINNED_ITEMS_QUERY_KEY, pagination], () => fetchFavoriteItems(pagination), {
+    onError: (errorThrown) => {
+      UserNotification.error(`Loading favorite items failed with status: ${errorThrown}`,
+        'Could not load favorite items');
+    },
+    retry: 0,
+    initialData: {
+      favorites: [],
+      ...DEFAULT_PAGINATION,
+    },
+  });
+};
+
+export const useRecentActivity = (pagination: RequestQuery): { data: PaginatedRecentActivity, isFetching: boolean} => {
+  return useQuery([RECENT_ACTIONS_QUERY_KEY, pagination], () => fetchRecentActivities(pagination), {
+    onError: (errorThrown) => {
+      UserNotification.error(`Loading recent activity failed with status: ${errorThrown}`,
+        'Could not load recent activity');
+    },
+    retry: 0,
+    initialData: {
+      recentActivity: [],
+      ...DEFAULT_PAGINATION,
+    },
+  });
+};
```

Comment:
Usually we have a `hooks` directory, let's keep the structure similar, or do you see a benefit in this structure?

Reply:
In this case, there are not many benefits of 1 file. I made that to have fewer files because we have only 3 hooks. I will separate them by files to keep the same structure